edxval/api.py

```diff
@@ -143,6 +149,87 @@ def update_video_status(edx_video_id, status):
     video.save()
 
 
+def get_video_transcript(video_id, lang_code):
+    """
+    Get a video's transcript
+
+    Arguments:
+        video_id: it can be an edx_video_id or an external_id extracted from external sources in a video component.
+        lang_code: it will the language code of the requested transcript.
+    """
+    try:
+        transcript = Transcript.objects.get(video_id=video_id, language=lang_code)
+    except Transcript.DoesNotExist:
+        transcript = None
+
+    return transcript
+
+
+def get_video_transcripts(video_id):
+    """
+    Get a video's transcripts
+
+    Arguments:
+        video_id: it can be an edx_video_id or an external_id extracted from external sources in a video component.
+    """
+    transcripts_set = Transcript.objects.filter(video_id=video_id)
+
+    transcripts = []
+    if transcripts_set.exists():
+        transcripts = TranscriptSerializer(transcripts_set, many=True).data
+
+    return transcripts
+
+
+def create_video_transcript(video_id, language, transcript_url, transcript_format, provider=TranscriptProviderType.CUSTOM):
+    """
+    Creates a transcript record for a video.
+
+    Arguments:
+        video_id: it can be an edx_video_id or an external_id extracted from external sources in a video component.
+        language: language code of a video transcript
+        transcript_url: url of a video transcript
+        transcript_format: format of the transcript
+        provider: transcript provider
+
+    Raises:
+        IntegrityError: raises IntegrityError if there is an existing transcript with same video_id and lang_code.
+    """
+    transcript = Transcript.objects.create(
+        video_id=video_id,
+        language=language,
+        transcript_url=transcript_url,
+        fmt=transcript_format,
+        provider=provider,
+    )
+
+    return TranscriptSerializer(transcript).data
+
+
+def update_video_transcript(video_id, language, transcript_url, transcript_format, provider=TranscriptProviderType.CUSTOM):
+    """
+    Update a transcript for a video
+
+    Arguments:
+        video_id: it can be an edx_video_id or an external_id extracted from external sources in a video component.
+        language: language code of a video transcript
+        transcript_url: url of a video transcript
+        transcript_format: format of the transcript
+        provider: transcript provider
+
+    Raises:
+        DoesNotExist: raises DoesNotExist if there is not any transcript for the given video_id and lang_code.
+    """
+    transcript = Transcript.objects.get(video_id=video_id, language=language)
+    transcript.language = language
+    transcript.transcript_url = transcript_url
+    transcript.fmt = transcript_format
+    transcript.provider = provider
+    transcript.save()
+
+    return TranscriptSerializer(transcript).data
```

Comment:
can we have 2 methods `get_or_create_transcripts` & `get_or_update_transcripts` as discussed.